src/domino.py

```diff
@@ -0,0 +1,208 @@
+            command.extend(['-mth', str(module_threshold)])
+
+        print('Running DOMINO with arguments: {}'.format(' '.join(command)), flush=True)
+
+        # container_framework = 'singularity' if singularity else 'docker'
+        domino_out = run_container(container_framework,
+                            'otjohnson/domino',
+                            command,
+                            volumes,
+                            work_dir)
+        print(domino_out)
+
+        ########
+
+        # Path(mapped_slices_dir).unlink(missing_ok=True)
+        # Path(out_dir, 'modules.out').unlink(missing_ok=True)
+        #for domino_output in out_dir.glob('modules.out'):
+        #    domino_output.unlink(missing_ok=True)
+
+        # domino creates a new folder in out_dir to output its modules files into /active_genes
+        out_modules_dir = Path(out_dir, 'active_genes')
+
+        # concatenate each module html file into one big file
+        with open(output_file, "w") as fo:
+            for html_file in out_modules_dir.glob('module_*.html'):
+                with open(html_file,'r') as fi:
+                    fo.write(fi.read())
+                # Path(html_file).unlink(missing_ok=True)
+
+
+    @staticmethod
+    def parse_output(raw_pathway_file, standardized_pathway_file):
+        """
+        Convert a predicted pathway into the universal format
+        @param raw_pathway_file: pathway file produced by an algorithm's run function
+        @param standardized_pathway_file: the same pathway written in the universal format
+        """
+        edges = pd.DataFrame()
+
+        with open(raw_pathway_file, 'r') as file:
+            for line in file:
+                if line.strip().startswith("let data = ["):
+                    line2 = line.replace('let data = ', '')
+                    line3 = line2.replace(';', '')
+
+                    data = json.loads(line3)
+
+                    entries = []
+                    for entry in data:
+                        tmp = entry['data']
+                        entries.append(tmp)
+
+                    df = pd.DataFrame(entries)
+                    newdf = df.loc[:,['source', 'target']].dropna()
+
+                    edges = pd.concat([edges, newdf], axis=0)
+
+        edges['rank'] = 1 # adds in a rank column of 1s because the edges are not ranked
+
+        # Remove the prefix
+        edges['source'] = edges['source'].apply(post_domino_id_transform)
+        edges['target'] = edges['target'].apply(post_domino_id_transform)
+
+        edges.to_csv(standardized_pathway_file, sep='\t',header=False, index=False)
+
+
+def pre_domino_id_transform(node_id):
+    """
+    Prepend each node id with ID_PREFIX
+    @param node_id: the node id to transformed
+    """
+    return ID_PREFIX + node_id
+
+
+def post_domino_id_transform(node_id):
+    """
+    Remove prefix
+    @param node_id: the node id to transformed
+    """
+    node_id = node_id[len(ID_PREFIX):]
```

Comment:
This works and will be fast. I want to think about whether we want to do a string replacement instead to protect against a node ID that doesn't have the expected prefix.

Reply:
If our inputted nodes were already Ensembl gene ids, would doing a string replacement of 'ENG0' replace both the prefix we manually added and the beginning 'ENG0' of the gene id?

Reply:
I wanted to use `removeprefix` but that wasn't added until Python 3.9 https://docs.python.org/3/library/stdtypes.html#str.removeprefix